internal/factsengine/gatherers/cibadmin_test.go

```diff
@@ -0,0 +1,145 @@
+package gatherers // nolint
+
+import (
+	"errors"
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/suite"
+	mocks "github.com/trento-project/agent/internal/factsengine/gatherers/mocks"
+)
+
+type CibAdminTestSuite struct {
+	suite.Suite
+	cibAdminOutput []byte
+}
+
+func TestCibAdminTestSuite(t *testing.T) {
+	suite.Run(t, new(CibAdminTestSuite))
+}
+
+func (suite *CibAdminTestSuite) SetupSuite() {
+	lFile, _ := os.Open("../../../test/fixtures/gatherers/cibadmin.xml")
+	content, _ := ioutil.ReadAll(lFile)
+
+	suite.cibAdminOutput = content
+}
+
+func (suite *CibAdminTestSuite) TestCibAdminGather() {
+	mockExecutor := new(mocks.CommandExecutor)
+
+	mockExecutor.On("Exec", "cibadmin", "--query", "--local").Return(
+		suite.cibAdminOutput, nil)
+
+	p := &CibAdminGatherer{
+		executor: mockExecutor,
+	}
+
+	factRequests := []FactRequest{
+		{
+			Name:     "instance_number",
+			Gatherer: "cibadmin",
+			Argument: "//primitive[@type='SAPHana']/instance_attributes/nvpair[@name='InstanceNumber']/@value",
```

Comment:
`//primitive[@type='SAPHana']/instance_attributes/nvpair[@name='InstanceNumber']/@value`
`//primitive[@type='SAPHana']/instance_attributes/nvpair[@name='SID']/@value`

this is what will be there in the DSL, I am wondering if we could simplify that by expecting something different, something like just the `instanceNumber` or `SID`, since the rest is actually the same :thinking: 

Anyway, I believe it is enough as a starting point, yet, I'd reason about options to simplify.

Reply:
I see also how that would require also accepting `type`, which opens to what @rtorrero raised recently about the possibility to provide support for multiple `Arguments`.

Reply:
Well, it turns out that these 2 examples are almost identical, but the xml output from cibadmin and crm_mon has a lot of paths. We couldn't really simply that much. At the end a xml output is a pretty generic thing, and the user would want to get any part of it.

Other examples:
```
//rcs_location[@node="node01"]/@rsc
//lrm_resource[@class="ocf]/lrm_rsc_op[@operation="monitor"]/@operation_key
```

At the end, the user would like to get any item from the xml. I think that trying to simply if would make the opposite, make it more difficult.
Finally, `xpath` is pretty well known standard (specially from the times where xml was more used), so I think keep in it as it is is the best solution

Reply:
I understand... let's just go further.